src/Hamiltonians/reduced_density_matrix.jl

```diff
@@ -344,3 +363,365 @@ function Base.iterate(od::TestOneParticleDensityOffdiagonals, state=(1, 1))
     end
     return nothing
 end
+
+"""
+     GradOneParticleDensity(v; normalize=true) <: AbstractOperator{SVector{m,T}}
+A one particle operator constructed from a provided test vector `v[1,:]` and 
+its gradient `v[2:m+1,:]` with respect to parameters `α₁,α₂,...,αₘ`. 
```

Comment:
Not sure how to read this. Should we pass the test vector and and its gradient as different parts of the same array? Is there a good reason for this? It seems like it would be much easier to understand (and handle) if you could pass separate objects.